### ISSUE TITLE
ReBar support 

### ISSUE THREAD

**Issue by narcyzzo** [KD-MM2/Qiyida-X99-D4-OC#1]
Hello, thanks for mods, can u add ReBar support for F8 bios?

**Comment by KD-MM2** [comment]
I will add mods for f8 bios later. or you can try following the guide here
https://www.youtube.com/watch?v=vcJDWMpxpjE&ab_channel=Miyconst

**Comment by narcyzzo** [comment]
Okay thank u. I can also confirm nvme slot works without any problem, bootable,  with F8 bios, only issue seems to be Ethernet, u think this is fixable by changing device paths in bios or some other modification?

**Comment by KD-MM2** [comment]
> Okay thank u. I can also confirm nvme slot works without any problem, bootable, with F8 bios, only issue seems to be Ethernet, u think this is fixable by changing device paths in bios or some other modification?

Wow nice, thanks for your confirmation.
I didn't have much experience or knowledge about EFI BIOS but I could see the different when comparing device path between f8's bios and stock bios using tools like MMTool, AMIBCP, RWEverything,..
![image](https://github.com/user-attachments/assets/edc6c456-32f1-4bc9-adcc-9c26d70af812)
In this image, we could see there is some option rom that exist in the stock but not in the f8 bios.
I tried to dump from stock bios then add to f8 bios but the device path accordingly are not exist(or missing) in the f8 bios.

![image](https://github.com/user-attachments/assets/2e9f9b8e-775d-4953-934d-a7df0f1e9b09)
![image](https://github.com/user-attachments/assets/034444ee-bbab-41c3-af98-0b656fb98acd)
![image](https://github.com/user-attachments/assets/c6837c39-f038-44ee-89b1-475d998b1fc1)
![image](https://github.com/user-attachments/assets/385e7b43-aef9-49d9-a09e-ada495e5d1d9)
![image](https://github.com/user-attachments/assets/3ff6c492-03a8-4c9f-a713-360b7cb6f1c4)
![image](https://github.com/user-attachments/assets/ba204d4a-c8b9-4fab-b512-538fc7b608ce)
![image](https://github.com/user-attachments/assets/697cfdb0-7a31-48e0-a09b-a3165c3aacc0)


And these images, we can see that stock bios have more 2 System slots than the f8 bios.

**Comment by narcyzzo** [comment]
Stock bios also has built in ReBar option, u think this can be ported to F8 bios in native way? without injtecting custom ReBarUEFI dxe like u posted in video

**Comment by KD-MM2** [comment]
> Stock bios also has built in ReBar option, u think this can be ported to F8 bios in native way? without injtecting custom ReBarUEFI dxe like u posted in video

I will check if the rebar dxe present and the option is hidden or something else, otherwise we must inject it.

**Comment by KD-MM2** [comment]
@narcyzzo 

the rebar dxe is present but there is no configuration entry. that means we must use tools(Windows) to enable rebar: [xCuri0/ReBarUEFI](https://github.com/xCuri0/ReBarUEFI)

another things, I found the new bios which more stable than current F8 bios, uploaded here:
https://github.com/KD-MM2/Qiyida-X99-D4-OC/blob/main/OpenCore-20241124/huananzhi-x99-tf-cx99de30.upt.rebar.rom.zip

I flashed bios image using external programmer so if you wanna try, do it with your own risk.
![image](https://github.com/user-attachments/assets/d8119986-1393-4fc8-88f3-89024b08ab73)
The third sata port works on this bios.

**Comment by KD-MM2** [comment]
![image](https://github.com/user-attachments/assets/d9b7deff-b1a8-409d-84e4-64c5be440e2d)

**Comment by narcyzzo** [comment]
@KD-MM2  what about this bios? https://github.com/Koshak1013/HuananzhiX99_BIOS_mods/tree/master/Huananzhi%20X99-F8/2020-05-25%20(%D1%80%D0%B5%D0%BA%D0%BE%D0%BC%D0%B5%D0%BD%D0%B4%D1%83%D0%B5%D1%82%D1%81%D1%8F)
![image](https://github.com/user-attachments/assets/03edabee-df7a-4dc5-8288-b1e53ab42ac7)
its same version u uploaded insie FPT zip? i says it has Rebar added

**Comment by narcyzzo** [comment]
its same version u uploaded insie FPT zip? i says it has Rebar added

**Comment by KD-MM2** [comment]
> There is any more adventages with this bios beside additional sata port working?
currently testing, seems more features(mods) compare with the current f8 bios.


> @KD-MM2 what about this bios? https://github.com/Koshak1013/HuananzhiX99_BIOS_mods/tree/master/Huananzhi%20X99-F8/2020-05-25%20(%D1%80%D0%B5%D0%BA%D0%BE%D0%BC%D0%B5%D0%BD%D0%B4%D1%83%D0%B5%D1%82%D1%81%D1%8F) ![image](https://private-user-images.githubusercontent.com/54555731/391340958-03edabee-df7a-4dc5-8288-b1e53ab42ac7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMwNjU1NTksIm5iZiI6MTczMzA2NTI1OSwicGF0aCI6Ii81NDU1NTczMS8zOTEzNDA5NTgtMDNlZGFiZWUtZGY3YS00ZGM1LTgyODgtYjFlNTNhYjQyYWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAxVDE1MDA1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBiMTc0YTIwOTk0MGUyYWVkZWU1YmUyYmZiNmM4OTA3NTgyNzQ5NTJmOWU1MzQyMDlmMWM0YTg3ODdjYjJiNDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5seiRCmqVzxXtoi7_mWusAOci3eoWb2te9h7r2lqQnc)

Not tested yet. If you have a programmer, you could try it.

> its same version u uploaded inside FPT zip? it says it has Rebar added

rebar dxe injected, i couldn't see the rebar option, but above 4g decoding option are showing.

**Comment by narcyzzo** [comment]
bios inside FPT zip is from what source?

**Comment by KD-MM2** [comment]
this: 
https://www.youtube.com/watch?v=3gus6gQE4gk
https://drive.google.com/file/d/16l6-NXeJc6TqOw0vVj0pGN7wrpTr58oF/view

**Comment by narcyzzo** [comment]
> this: https://www.youtube.com/watch?v=3gus6gQE4gk https://drive.google.com/file/d/16l6-NXeJc6TqOw0vVj0pGN7wrpTr58oF/view
This bios i same as this 

> ![image](https://private-user-images.githubusercontent.com/54555731/391340958-03edabee-df7a-4dc5-8288-b1e53ab42ac7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMxNzgyNDgsIm5iZiI6MTczMzE3Nzk0OCwicGF0aCI6Ii81NDU1NTczMS8zOTEzNDA5NTgtMDNlZGFiZWUtZGY3YS00ZGM1LTgyODgtYjFlNTNhYjQyYWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAyVDIyMTkwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3ZTU0NDk5NTA1MGIwN2JiNzkyZWRhYTcwMWI1NDg4MzI5MzZiZTE2NTUyY2UzZTgyMzJkYTQxMDdmMjc2NzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xht5yPY_DnLWgBw5poNM_mtAyGO1Tz62Lhxae_gwVtE)

**Comment by narcyzzo** [comment]
> this: https://www.youtube.com/watch?v=3gus6gQE4gk https://drive.google.com/file/d/16l6-NXeJc6TqOw0vVj0pGN7wrpTr58oF/view
> > This bios i same as this
> 
> > ![image](https://private-user-images.githubusercontent.com/54555731/391340958-03edabee-df7a-4dc5-8288-b1e53ab42ac7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMxNzgyNDgsIm5iZiI6MTczMzE3Nzk0OCwicGF0aCI6Ii81NDU1NTczMS8zOTEzNDA5NTgtMDNlZGFiZWUtZGY3YS00ZGM1LTgyODgtYjFlNTNhYjQyYWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAyVDIyMTkwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3ZTU0NDk5NTA1MGIwN2JiNzkyZWRhYTcwMWI1NDg4MzI5MzZiZTE2NTUyY2UzZTgyMzJkYTQxMDdmMjc2NzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xht5yPY_DnLWgBw5poNM_mtAyGO1Tz62Lhxae_gwVtE)

**Comment by KD-MM2** [comment]
Thanks for confirm that. Beside bios mod for x99d4, this repository also contains opencore efi for hackintosh. do you have any interest ?

**Comment by narcyzzo** [comment]
Yes, i have a lot experience in hackintoshing and will be doing that but firstly i mod bios and OC Cpu

**Comment by KD-MM2** [comment]
WOW nice. Waiting for that. I'm using ventura because intel wlan kext not stable on sequoia.

**Comment by narcyzzo** [comment]
if u have windows, can u check if virtualization technology VTX(VMX) in this bios is working? Since i have it enabled and in windows says its not enabled so something is broken here

**Comment by narcyzzo** [comment]
@KD-MM2

**Comment by KD-MM2** [comment]
@narcyzzo sorry, work's deadlines are coming in the year end. I don't have much free time to confirm that.

> if u have windows, can u check if virtualization technology VTX(VMX) in this bios is working? Since i have it enabled and in windows says its not enabled so something is broken here

**Comment by jwagnervaz** [comment]
The fix for all you documented here is my port.

https://github.com/jwagnervaz/QIYIDA-X99-D4-V2.0

**Comment by KD-MM2** [comment]
> The fix for all you documented here is my port.
> 
> https://github.com/jwagnervaz/QIYIDA-X99-D4-V2.0

nice, i will try it later.

**Comment by jwagnervaz** [comment]
> [@jwagnervaz](https://github.com/jwagnervaz) Appreciated that. Have you tried to boot opencore yet(assuming you already did)? this opencore efi(repo)? Later(maybe right tonight - ~22:00 gmt+9) I will try your bios and rebuild an EFI for this board.
> 
> I know this is kind of an unreasonable request, but could you share a bit of experience or guidance on how to mod/port a BIOS? I just found a Machinist X99 PR9 at a local second-hand shop and I’m planning to mess around with its BIOS. (Of course, I’ve got all the necessary gear like a flasher, soldering + hot air station, and about a dozen spare SPI flash chips, so no worries about me accidentally bricking the BIOS chip.)

Yes i did try opencore, you have to rebuil from strach your efi, cause ssdt has changed in my bios with a lot of fixes, its working, use [OP](https://github.com/lzhoang2801/OpCore-Simplify) it makes the efi auto, and dump the ssdt, if you wanna do in a manual way feals free. 

Machinist X99 PR9

give me the origirinal dump and i will port my bios for it, in seconds, i have x99 build tree now, but its intel for intel, so i cant share.
i can port to any x99 c612 chinese im minutes now, im planing to port a grafical bios next. 

share your experience with my bios so i can improve in a next version.

**Comment by jwagnervaz** [comment]
@KD-MM2 https://discord.gg/dsnTGTGW join at, we can colaborate if you want, i have some ideias to make our skills improved and profit on this path.

**Comment by KD-MM2** [comment]
> > [@jwagnervaz](https://github.com/jwagnervaz) Appreciated that. Have you tried to boot opencore yet(assuming you already did)? this opencore efi(repo)? Later(maybe right tonight - ~22:00 gmt+9) I will try your bios and rebuild an EFI for this board.
> > I know this is kind of an unreasonable request, but could you share a bit of experience or guidance on how to mod/port a BIOS? I just found a Machinist X99 PR9 at a local second-hand shop and I’m planning to mess around with its BIOS. (Of course, I’ve got all the necessary gear like a flasher, soldering + hot air station, and about a dozen spare SPI flash chips, so no worries about me accidentally bricking the BIOS chip.)
> 
> Yes i did try opencore, you have to rebuil from strach your efi, cause ssdt has changed in my bios with a lot of fixes, its working, use [OP](https://github.com/lzhoang2801/OpCore-Simplify) it makes the efi auto, and dump the ssdt, if you wanna do in a manual way feals free.
> 
> Machinist X99 PR9
> 
> give me the origirinal dump and i will port my bios for it, in seconds, i have x99 build tree now, but its intel for intel, so i cant share. i can port to any x99 c612 chinese im minutes now, im planing to port a grafical bios next.
> 
> share your experience with my bios so i can improve in a next version.

wow so impressive, it's something like you are building the bios image from scratch(source/tree)?
I was thinking about it before when I bought the x99 board but since I'm insufficient of time so I gave it up.

**Comment by KD-MM2** [comment]
> [@KD-MM2](https://github.com/KD-MM2) https://discord.gg/dsnTGTGW join at, we can colaborate if you want, i have some ideias to make our skills improved and profit on this path.

joined :D

**Comment by narcyzzo** [comment]
hello @jwagnervaz windows11 is broken with your bios, blue screen and freezes, reboots, event in safe mode

**Comment by narcyzzo** [comment]
ey @KD-MM2  where is your little discord cunt friend who is kicking ppl who are tells the truth about his shitty broken bios? lmao

**Comment by jwagnervaz** [comment]
1- Didn't invite you @narcyzzo 

2- You are too much inteligent to discuss with.

3- I didn't get payed to give you support.

4- as holes are not welcome

5- the bug your provoked using a different opencore build (f8 bios) to boot with windows 

6- as you so smart do your own build bios.

7- people like you dont  have friends

8- rude people ar not allowed here at github or iven discord. 

9- your are the hacker master.

10- return  to 1st reason.

**Comment by KD-MM2** [comment]
> ey [@KD-MM2](https://github.com/KD-MM2) where is your little discord cunt friend who is kicking ppl who are tells the truth about his shitty broken bios? lmao

Don't be rude like that, be a smart internet user.
We are not paid for these things nor do we have any responsibility for you.
Remember, everything you do on the internet -> DO IT WITH YOUR OWN RISKS.
Before complaining, why not try:
- Resetting BIOS/NVRAM settings using the jumper on the motherboard.
- Try disabling secure boot/switch it to standard mode.
- For OpenCore, please follow the configuration instructions: [OpenCore-20250927/BIOS.md](https://github.com/KD-MM2/Qiyida-X99-D4-OC/blob/main/OpenCore-20250927/BIOS.md)
- Remember, every action when changing BIOS fw *must* be accompanied by resetting the settings.

I have used it for almost a week and haven't seen any issues. The problem is you.
I am very appreciative that @jwagnervaz contributed this BIOS for us. Without him, we wouldn't have this BIOS and would be stuck with the buggy F8 BIOS. 

So please respect that.

**Comment by KD-MM2** [comment]
@narcyzzo 

I've seen your messages on Discord. That's not our problem though.
- If you are getting BSOD issue, that's the driver problem.
- If there's BIOS problem, it MUST BE BSOD right after the loading animation appears.
- How could you think you can go without reinstall the driver for the new bios?(PCI Device Path, ACPI,... and a tons of things to be difference)
- https://www.3dpchip.com/3dpchip/index_eng.html
- @jwagnervaz  also sent a video about booting windows 11 without issues, so that's not problem.
- Remember when dual-booting opencore(macos) with windows, DO NOT ENTER WINDOWS USING OPENCORE, IT MUST BE ENTERED FROM THE BOOT MENU(F11).
- (The reason why?) If you enter windows using opencore, these got patched(even there is os check in acpi patches but at least it got affected) and becoming weird in windows.

**Comment by narcyzzo** [comment]
dude, im tested booting without opencore, and same result, bsod happen random,  sometimes when windows loading, sometims right after boot, in safe mode, when windows start repairing itself also. This dude must be so much retarded to kick someone who talking bad about his work, being rude and think he is the bios god, dont want to help and blocking random user in internet. Windows works without reinstalling when changing motherboards, bioses and any other changes, i tested this many times and there is much info about it everywhere. But yeah this is fcked chinese board so many strange things can happen. The last thing i will check is clean install, but when it bsod when loading, in safe mode also where drivers are not loaded, its not optimistic

**Comment by narcyzzo** [comment]
and talking about opencore is the problem is so bulshit, if everything is correctly configured, dualbot works via opencore always, as it works with F8 bios before dude, and on many other hackintoshes i made, so bad excuse again master of bioses, Its better to stop spread misinformation, and coperate with people who use your work, to make it better or best.

**Comment by KD-MM2** [comment]
@narcyzzo 

Me and the author are using that bios without any issues. You may want to check against your "clean install" method.
